File: python-bindings/tests/wrye_report_conformance.py
"""Input-only Wrye HTML parsing and exact public formatter observations."""

from collections.abc import Mapping
from typing import Any


def observe_wrye_report(fixture: Mapping[str, Any]) -> dict[str, Any]:
    """Project every native issue field and compare the public convenience formatter."""
    import classic_scangame as native

    parser = native.WryeBashParser(fixture["warnings"])
    issues = parser.parse(fixture["html"])
    report = parser.format_report(issues)
    if native.parse_wrye_report(fixture["html"], fixture["warnings"]) != report:
        raise ValueError("Wrye convenience parser changed formatted report")
    return {
        "issues": [
            {
                "section": issue.section_title,
                "plugins": issue.plugins,
                "warning": issue.warning_message,
                "severity": str(issue.severity).split(".")[-1],
            }
            for issue in issues
        ],
        "report": report,
    }

File: node-bindings/classic-node/__test__/wrye_report_conformance.ts
import {JsWryeBashParser} from "../index.js";

/** Project actual native parser fields and the complete formatted report string. */
export function observeWryeReport(fixture: Record<string, any>): Record<string, any> {
    const parser = new JsWryeBashParser(fixture.warnings);
    const issues = parser.parse(fixture.html);
    return {
        issues: issues.map(issue => ({
            section: issue.sectionTitle, plugins: issue.plugins,
            warning: issue.warningMessage ?? null, severity: issue.severity
        })),
        report: JsWryeBashParser.formatReport(issues)
    };
}
